test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/AssociatedMetadataProviderTest.cs

```diff
@@ -233,6 +233,42 @@ public void GetMetadataForProperties_SetsContainerAsExpected()
             Assert.Same(model, metadata[2].Container);
         }
 
+        [Fact]
+        public void GetMetadataForPropertiesFromModelMetadataType()
```

Comment:
- expectations should be obvious from test name.  change to something like `GetMetadataForProperties_IncludesModelMetadataTypeAtributes`
- also test `GetMetadataForProperty()` and `GetMetadataForType()`